api/xapi/helpers/guards.rb

```diff
@@ -0,0 +1,11 @@
+module Xapi
+  module Helpers
+    module Guards
+      def require_key
+        return if params[:key]
```

Comment:
possibly consider? 

```ruby
params.fetch(:key) { halt 401, { error: "Please provide your Exercism.io API key" }.to_json }
```
not sure though if this would be as readable. 😄